docs/porting/custom-target-porting.md

```diff
@@ -0,0 +1,258 @@
+
+1. [Install Mbed CLI](../tools/installation-and-setup.html) if you don't already have it.
+
+1. (Optional) Create a new Mbed program (for example, `mbed-os-imaginary-port`).
+
+   If you don't already have an Mbed program on your computer, run this Mbed CLI command in a command terminal:
+
+   ```
+   mbed new --program mbed-os-imaginary-port
+   ```
+   
+   This command creates a new program folder called `mbed-os-imaginary-port` and then imports `mbed-os` from the [official Mbed OS source repository](https://github.com/armmbed/mbed-os) into it.
+
+1. Change directories into your new project:
+
+  ```
+  cd mbed-os-imaginary-port
+  ```
+
+1. Create a new file named `custom_targets.json` at the same level as the `mbed-os` directory.
+
+1. Inspect the contents of `mbed-os/targets/targets.json`. For this example, search for `DISCO_L475VG_IOT01A`.
+
+1. Copy the contents from the `DISCO_L475VG_IOT01A` section into your `custom_targets.json` file. Be sure to include brackets `{ }` surrounding the content.
+
+### Customizing
+
+1. Make changes to `custom_targets.json` for your board. For example, after making changes, the full contents look like this:
+   
+   ```
+   {
+     "IMAGINARYBOARD": {
+         "components_add": ["QSPIF", "FLASHIAP"],
+         "inherits": ["FAMILY_STM32"],
+         "core": "Cortex-M4F",
+         "extra_labels_add": ["STM32L4", "STM32L475xG", "STM32L475VG"],
+         "config": {
+             "clock_source": {
+                 "help": "Mask value : USE_PLL_HSE_EXTC (need HW patch) | USE_PLL_HSE_XTAL (need HW patch) | USE_PLL_HSI | USE_PLL_MSI",
+                 "value": "USE_PLL_MSI",
+                 "macro_name": "CLOCK_SOURCE"
+             },
+             "lpticker_lptim": {
+                 "help": "This target supports LPTIM. Set value 1 to use LPTIM for LPTICKER, or 0 to use RTC wakeup timer",
+                 "value": 1
+             }
+         },
+         "overrides": { "lpticker_delay_ticks": 4 },
+         "detect_code": ["1234"],
+         "macros_add": [
+             "MBED_TICKLESS",
+             "MBED_SPLIT_HEAP"
+         ],
+         "device_has_add": [
+             "CRC",
+             "TRNG",
+             "FLASH",
+             "QSPI",
+             "MPU"
+         ],
+         "device_has_remove": [         
+             "ANALOGIN",
+             "I2CSLAVE",
+             "I2C_ASYNCH"
+         ],                     
+         "release_versions": ["2", "5"],
+         "device_name": "STM32L475VG",
+         "bootloader_supported": true
+     }
+   }
+   ```
+
+   Let's review the changes one by one.
+
+#### Changes
+
+1. The board name changed from `DISCO_L475VG_IOT01A` to `IMAGINARYBOARD`, so the board can be uniquely identified.
+1. The `detect_code` changed from `0764` to `1234`. The `detect_code` is a unique number that identifies the board to the Mbed OS test tools. For Mbed Enabled boards, this number is exposed through the debug interface with Mbed CLI by typing `mbedls`.
```

Comment:
Do we need some information how to set it? When do you need to get it from us and when it's ok to just pick some? How to avoid clashes? Consideration based on public vs private port? I guess it deserves a link to a separate doc.

Reply:
Looking through our documentation - I see it mentioned a few times, but never really see it explained well and the terminology is inconsistent.  

- Sometimes its called a **board id** (https://www.mbed.com/en/about-mbed/mbed-enabled/requirements/).  
- Sometimes it is called **four-digit hexadecimal identifier** (https://os.mbed.com/docs/mbed-os/v5.14/porting/mbed-ls.html)
- Sometimes its called a **Platform ID** (https://github.com/ARMmbed/mbed-os-tools/tree/master/packages/mbed-ls#adding-platform-support).  
- Sometimes its called **unique board ID** (https://os.mbed.com/docs/mbed-os/v5.14/porting/setting-up.html)

I like Board ID, but Platform ID or the others seem fine.  

For me the key points are: 
1. Board IDs are unique identifiers assigned by the Arm Mbed team.
2. Board IDs are used by the Mbed tools to identify the connected board.
3. You really only need your own unique ID if you plan to push your platform to the Mbed community.
4. You can make Mbed CLI recognize your own custom board even without getting your own Board ID.
5. Contact a technical account manger or Mbed support if you would like to request a board ID.  

@AnotherButler - can you recommend where this information would be best located?

Reply:
I just added a new commit for this PR with changes related to the Platform ID. Also, I made pull request https://github.com/ARMmbed/mbed-os-5-docs/pull/1175 to align the terminology and add more detail.  Please let me know if this addresses the comments.